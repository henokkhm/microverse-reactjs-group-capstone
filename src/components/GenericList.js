// eslint-disable-next-line import/no-extraneous-dependencies
import PropTypes from 'prop-types';
import styles from '../styles/GenericList.module.css';

function GenericList({ data }) {
  return (
    <ul className={styles.list}>
      {data.map((item) => (
        <li key={item.id} className={styles.listItem}>
          {item.value}
        </li>
      ))}
    </ul>
  );
}

GenericList.propTypes = {
  data: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired,
    }),
  ).isRequired,
};

export default GenericList;
