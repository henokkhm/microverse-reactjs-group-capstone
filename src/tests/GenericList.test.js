import { render } from '@testing-library/react';

import GenericList from '../components/GenericList';

it('The GenericList component renders the list provided as data correctly', () => {
  const data = [
    {
      id: 'first_id',
      value: 'first_value',
    },
    {
      id: 'second_id',
      value: 'second_value',
    },
  ];
  const { asFragment } = render(<GenericList data={data} />);
  expect(asFragment()).toMatchSnapshot();
});
