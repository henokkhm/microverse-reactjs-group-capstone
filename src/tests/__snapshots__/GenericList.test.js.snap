// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The GenericList component renders the list provided as data correctly 1`] = `
<DocumentFragment>
  <ul
    class="list"
  >
    <li
      class="listItem"
    >
      first_value
    </li>
    <li
      class="listItem"
    >
      second_value
    </li>
  </ul>
</DocumentFragment>
`;
